### PR TITLE
Made fixes to issues which regard incorrect API response/request hand…

### LINKED ISSUE #168: API Contract Issues
Integration Issues: API Contract

1. DiscussionEndDTO
* playerisKilled - should be playerOnTrial
* Pass in nickname to the DTO playerChosen.nickname BUT ONLY if playerChosen is not null (if its null we pass in null)

2. NightStateChange Events 
* Comment out the eventscript in NightStateChangeEvents.js

3. Block dead people from voting in the frontend (can be fixed on frontend)

4. Sending nicknames of winners. Currently the player object is being sent
* Set of players.map ((p) => p.nickname))

5. isGameOverDTO
* Incorrect field names being sent from the backend

### CheckList:
- [x] Error 1
- [x] Error 2
- [x] Error 3
- [x] Error 4
- [x] Error 5

### Issue Approval:
- [x]  Issue approved by: @ateagit

### PR DESCRIPTION
**Pull request by MKTPR**
# Description
This PR fixes #168 

DiscussionEnd DTO and GameOverDTO has been changed to meet the API response/request contract

Player objects in requests have been changed to nicknames to meet the contract.

# Type of change:
* Bug fix (Non-breaking change which fixes a bug)
* Breaking change (This change would cause existing functionality to break)

# Checklist: 
* Have you merged main into your branch?
* Have you tested your changes to ensure it works as expected and does not break existing functionality?
* If applicable, please ensure sufficient tests are added that is related to the changes.
* Please ensure this PR has a `label`, is linked to an `issue` and is related to a `project`
* If new documentation is required for this change, have you created a new documentation issue that describes the documentation needed?

### CODE REVIEW

File: backend/domain/DTO/response/DiscussionEndDTO.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class DiscussionEndDTO {
-    constructor(playerKilled) {
-        this.playerKilled = playerKilled;
+    constructor(playersOnTrial) {
+        this.playersOnTrial = playersOnTrial;
```
Comment (parfei) [inline backend/domain/DTO/response/DiscussionEndDTO.js:3]: Nitpick: should be `playerOnTrial` right? Since we simplified the logic down.

**Review by parfei — APPROVED** [review]
:D

**Review by reeved — APPROVED** [review]
LGTM

File: backend/Events/GameStateEvents/NightStateChangeEvents.js [hunk 2]
```diff
@@ -47,9 +45,14 @@ function endNight(io, socket, mafiaGame) {
 
     const winningRole = room.getWinningRole();
     if (winningRole !== null) {
-        io.in(roomID).emit('game-over', new GameOverDTO(winningRole, room.getPlayersByRole(winningRole)));
+        io.in(roomID).emit(
+            'game-over',
+            new GameOverDTO(
+                winningRole,
+                room.getPlayersByRole(winningRole).map((p) => p.nickname)
```
Comment (Dhors) [inline backend/Events/GameStateEvents/NightStateChangeEvents.js:52]: did we fix this one nonbreaking bug yet? 
(where winning role said civilian and it didn't include detective or medic)

File: backend/domain/DTO/response/DiscussionEndDTO.js (hunk 1 above)
Comment (MKTPR) [reply backend/domain/DTO/response/DiscussionEndDTO.js:3]: could we possibly leave it as playersOnTrial for future implementation? If not it should fine to get rid of it as well!

**Review by Dhors — APPROVED** [review]
as long as we are getting the right winning players, otherwise looks good to me!

Just tested: everything works fine. Good work Maric!

File: backend/Events/GameStateEvents/NightStateChangeEvents.js (hunk 2 above)
Comment (MKTPR) [reply backend/Events/GameStateEvents/NightStateChangeEvents.js:52]: oh we fixed it and it'll be included in the next PR!

File: backend/domain/DTO/response/DiscussionEndDTO.js (hunk 1 above)
Comment (parfei) [reply backend/domain/DTO/response/DiscussionEndDTO.js:3]: Yeah, we could rename it for this instance as people could get confused while working on this :) If we do allow multiple players to go on trial, we can rename the variable later!

**Pull request merged**